EmbedSample/Views/Dashboard/Report.cshtml

```diff
@@ -3,9 +3,180 @@
+                    if (pages.length === 0) {
+                        return;
+                    }
+
+                    pageIndex = pages.findIndex(function (el) {
+                        return el.name === e.detail.newPage.name;
+                    });
+                });
+
+            // For a complete guide to page navigation see the following wiki page
+            // https://github.com/Microsoft/PowerBI-JavaScript/wiki/Page-Navigation
+            function changePage(direction) {
+                var nextPageIndex = pageIndex + direction;
+                if (nextPageIndex < 0) nextPageIndex = pages.length - 1;
+                if (nextPageIndex >= pages.length) nextPageIndex = 0;
+
+                pages[nextPageIndex].setActive();
+            }
+
+            // For a complete guide to setting filters see the following wiki page
+            // https://github.com/Microsoft/PowerBI-JavaScript/wiki/Filters
+            function addFilter() {
+                var target = $('#filter-target').val();
+                var table = $('#filter-table').val();
+                var column = $('#filter-column').val();
+                var value = $('#filter-value').val();
+
+                var basicFilter = {
+                    $schema: "http://powerbi.com/product/schema#basic",
+                    target: {
+                        table: table,
+                        column: column
+                    },
+                    operator: 'In',
+                    values: [value]
+                };
+
+                var filterTarget = target === 'page' ? currentPage : report;
+                // Get existing filters and append a new filter
+                // https://microsoft.github.io/PowerBI-JavaScript/interfaces/_src_ifilterable_.ifilterable.html#getfilters
+                filterTarget.getFilters().then(function (allTargetFilters) {
+                    allTargetFilters.push(basicFilter);
+
+                    // Set filters
+                    // https://microsoft.github.io/PowerBI-JavaScript/interfaces/_src_ifilterable_.ifilterable.html#setfilters
+                    filterTarget.setFilters(allTargetFilters);
+                });
+
+
+                $('#filter-form')[0].reset();
+            }
+
+            function clearFilters() {
+                // Remove report filters
+                // https://microsoft.github.io/PowerBI-JavaScript/classes/_src_report_.report.html#removefilters
+                report.removeFilters();
+                pages.forEach(function (page) {
+                    // Remove page filters
+                    // https://microsoft.github.io/PowerBI-JavaScript/classes/_src_page_.page.html#removefilters
+                    page.removeFilters();
+                });
+            }
+
+            // For a full list of configurable settings see the following wiki page
+            // https://github.com/Microsoft/PowerBI-JavaScript/wiki/Settings
+            function updateSetting(e, settingName) {
```

Comment:
Not required, but I think the abstraction should be that this function doesn't have to know about events. It should just get the setting name and the boolean directly. 

Instead of passing in the event here, it should take the `event.target.checked` from calling function on event handler.
This keeps event handler concerned with events and business logic here independent of DOM/html